src/Gui/ElideCheckBox.h

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2025 Alfredo Monclus <alfredomonclus@gmail.com>          *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+// This custom widget adds the missing ellipsize functionality in QT5
+
+#ifndef ELIDECHECKBOX_H
+#define ELIDECHECKBOX_H
+
+#ifndef _PreComp_
+#include <QCheckBox>
+#include <QPainter>
+#include <QFontMetrics>
+#include <QStyleOptionButton>
+#endif
+
+
+class GuiExport ElideCheckBox : public QCheckBox {
```

Comment:
Sorry, forgot to mention that the `GuiExport` macro is defined in FCGlobal .h, so you'll need to include that.